pkg/apply/prune/inventorycm.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package prune
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+func WrapInventoryObj(info *resource.Info) Inventory {
+	return &InventoryConfigMap{inv: info}
+}
+
+type InventoryConfigMap struct {
+	inv      *resource.Info
```

Comment:
Do we need these implementations to hold state? I was thinking it would be cleaner if we kept whatever state we need in the applier, and just call out to the inventory implementation whenever the code is specific to the type.

Reply:
I don't understand this. Let's talk in person.